web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java

```diff
@@ -24,28 +24,49 @@
 import org.springframework.util.Assert;
 
 /**
- * A logout handler which clears a defined list of cookies, using the context path as the
- * cookie path.
+ * A logout handler which clears either
+ * - A defined list of cookie names, using the context path as the cookie path
+ * OR
+ * - A given list of Cookies
  *
  * @author Luke Taylor
  * @since 3.1
  */
 public final class CookieClearingLogoutHandler implements LogoutHandler {
-	private final List<String> cookiesToClear;
+	private final List<Object> cookiesToClear;
```

Comment:
Let's change this to `List<Function<HttpServletRequest,Cookie>>`. In each of the constructors we can convert the arguments into a `Function<HttpServletRequest,Cookie>`.